libsolidity/AST.cpp

```diff
@@ -342,8 +347,11 @@ void VariableDeclaration::checkTypeRequirements()
 	if (!m_value)
 		return;
 	if (m_type)
+	{
 		m_value->expectType(*m_type);
-	else
+		if (m_isStateVariable && !m_type->externalType() && getVisibility() >= Visibility::Public)
```

Comment:
It is actually more complicated here: You have to create the function type of the accessor (by using the FunctionType constructor which takes a variable declaration as argument) and check that its type is compatible with external visibility.